metadataserver/metatron/sign.go

```diff
@@ -0,0 +1,96 @@
+package metatron
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"fmt"
+	"path/filepath"
+
+	"github.com/Netflix/titus-executor/api/netflix/titus"
+)
+
+const (
+	// Matches java.security.spec.PSSParameterSpec.DEFAULT.getSaltLen()
+	saltLength      = 20
+	defaultCertPath = "/run/metatron/certificates"
+)
+
+// Signer signs bytes using an x509 cert / key pair: in practice, this is the Metatron client cert
+type Signer struct {
+	Algorithm   string
+	Certificate tls.Certificate
+	Hash        crypto.Hash
+	SignerOpts  crypto.SignerOpts
+	Signer      crypto.Signer
+}
+
+// NewDefaultSigner creates a signer with the default certificate path to the Metatron client cert
+func NewDefaultSigner() (*Signer, error) {
+	cert, err := tls.LoadX509KeyPair(filepath.Join(defaultCertPath, "client.crt"), filepath.Join(defaultCertPath, "client.key"))
+	if err != nil {
+		return nil, err
+	}
+
+	return NewSigner(cert)
+}
+
+// NewSigner creates a new signer with the passed in TLS certificate
+func NewSigner(cert tls.Certificate) (*Signer, error) {
```

Comment:
@douglasc-nflx / @JackOfMostTrades - Can you review this to make sure I'm not doing anything dumb?  Tests are in `metadataserver/server_test.go`.